### PR TITLE
[TIMOB-15910] Android: Implement Ti.UI.Android.DrawerLayout

### PR DESCRIPTION
**Pull request by garymathews**
- Based on community module https://github.com/manumaticx/Ti.DrawerLayout
- Implement `Titanium.UI.Android.DrawerLayout`

##### TEST CASE
```JS
var win = Ti.UI.createWindow(),
    leftView = Ti.UI.createView({backgroundColor:'red'}),
    centerView = Ti.UI.createView({backgroundColor:'yellow'}),
    rightView = Ti.UI.createView({backgroundColor:'orange'}),
    drawer = Ti.UI.Android.createDrawerLayout({
        leftView: leftView,
        centerView: centerView,
        rightView: rightView
    }),
    btn = Ti.UI.createButton({title: 'RIGHT'});

btn.addEventListener('click', function() {
    drawer.toggleRight();
});

centerView.add(btn);

win.addEventListener('open', function(){
    var activity = win.getActivity(),
        actionbar = activity.getActionBar();

    if (actionbar) {
        actionbar.displayHomeAsUp = true;
        actionbar.onHomeIconItemSelected = function() {
            drawer.toggleLeft();
        };
    }
});

win.add(drawer);
win.open();
```

[JIRA Ticket](https://jira.appcelerator.org/browse/TIMOB-15910)

### CODE REVIEW

File: android/titanium/src/java/org/appcelerator/titanium/TiC.java [hunk 1]
```diff
@@ -576,6 +576,21 @@
...
+
+	/**
+	 * @module.api
+	 */
+	public static final String EVENT_DRAWER_OPEN = "draweropen";
+
+	/**
+	 * @module.api
+	 */
+	public static final String EVENT_DRAWER_SLIDE = "drawerslide";
```
Comment (hansemannn) [inline android/titanium/src/java/org/appcelerator/titanium/TiC.java]: Can we name it `open`, `close` and `slide` instead? Feels more unified with other related API's. Thx!

**Comment by jquick-axway** [comment]
@garymathews, I have an interesting idea.

I suspect that many Titanium developers may want to implement a tab group like feature via your new navigation drawer feature since that type of interface is commonly implemented on Android. _Theoretically_, we should be able to do this now by using a `Ti.UI.TabGroup` with the tabs hidden via the `TabGroup.disableTabNavigation(true)` method... and when a DrawerLayout button has been tapped, it can display one of the tabs via code. If we can prove that this will work, then it'll be worth documenting. What do you think?

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIDrawerLayout.java [hunk 2]
```diff
@@ -0,0 +1,577 @@
...
+
+import android.support.v4.widget.DrawerLayout;
+import android.support.v4.widget.DrawerLayout.LayoutParams;
+import android.support.v4.widget.ViewDragHelper;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.app.ActionBarDrawerToggle;
+import android.support.v7.widget.Toolbar;
+import android.util.Log;
+import android.view.*;
+import android.widget.FrameLayout;
```
Comment (ypbnv) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIDrawerLayout.java]: Remove wildcard import.

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIDrawerLayout.java [hunk 3]
```diff
@@ -0,0 +1,577 @@
...
+import org.appcelerator.titanium.view.TiUIView;
+
+import android.support.v4.widget.DrawerLayout;
+import android.support.v4.widget.DrawerLayout.LayoutParams;
+import android.support.v4.widget.ViewDragHelper;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.app.ActionBarDrawerToggle;
+import android.support.v7.widget.Toolbar;
+import android.util.Log;
+import android.view.*;
```
Comment (ypbnv) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUIDrawerLayout.java]: Remove wildcard import.

**Review by ypbnv — CHANGES_REQUESTED** [review]
Besides the import everything else looks good.

**Comment by lokeshchdhry** [comment]
FR Passed.

The drawer layout works as expected. The left, middle & the right views are shown with the appropriate way of unhiding them. Also, different views like imageView, Label, TextView, tableview, listview etc can be successfully added to the left, middle & right drawer views.
The properties & methods seem to work as expected.

Studio Ver:     4.9.1.201707200100
SDK Ver:        7.0.0 Local build
OS Ver:         10.12.3
Xcode Ver:      Xcode 8.3.3
Appc NPM:       4.2.9
Appc CLI:       6.2.3
Ti CLI Ver:     5.0.14
Alloy Ver:      1.9.13
Node Ver:       6.10.1
Java Ver:       1.8.0_101
Devices:        ⇨ google Nexus 5 --- Android 6.0.1
                      ⇨ google Pixel --- Android 7.1.1
Emulator:     Android 4.4.2

**Pull request merged**